### PR TITLE
Deployment record improvements

### LINKED ISSUE #820: Re-deploying old app fails because of `envVars`
```
Error in `deploymentRecord()` at rsconnect/R/deployments.R:93:2:
! `envVars` must be a character vector or `NULL`, not `NA`.
Run `rlang::last_trace()` to see where the error occurred.
```

### LINKED ISSUE #821: Version the deployment dcf record
Should do it at as a matter of principle, and it will be useful for the upcoming app ID -> content ID change for cloud.

### PR DESCRIPTION
**Pull request by hadley**
* Improve `envVars` handling. Fixes #820.
* Add support for `version`. Fixes #821.
* Name arguments to `createDeploymentTarget()` calls

### CODE REVIEW

File: R/deployments.R [hunk 1]
```diff
@@ -79,9 +80,11 @@ deployments <- function(appPath = ".",
 
 deploymentFields <- c(
   "name", "title", "username", "account", "server", "hostUrl", "appId",
-  "bundleId", "url", "envVars"
+  "bundleId", "url", "envVars", "version"
 )
 
+deploymentVersion <- 1L
```
Comment (hadley) [inline R/deployments.R:86]: Maybe this should be `deploymentRecordVersion`?

Comment (aronatkins) [reply R/deployments.R:86]: I like that better.

File: tests/testthat/test-deployments.R [hunk 2]
```diff
@@ -69,6 +69,14 @@ test_that("can read/write metadata", {
   expect_equal(out$meta2, "two")
 })
 
+test_that("can read/write version", {
```
Comment (aronatkins) [inline tests/testthat/test-deployments.R:72]: Where do we test reading a DCF file that lacks the `envVars` and `version` fields?

We should test reading single files with/without these fields as well as a collection of files where some of them lack these fields.

**Review by aronatkins — APPROVED** [review]
The `envVars` and `version` fields are new and should not cause much difficulty if folks roll-back to an old release; old rsconnect will parse those fields, but may remove them on write.

Should we add something to NEWS about the consequences to stored `envVars` if folks upgrade then downgrade?

https://github.com/rstudio/rsconnect/blob/v0.8.29/R/deployments.R

File: tests/testthat/test-deployments.R [hunk 3]
```diff
@@ -80,10 +88,15 @@ test_that("can read/write env vars", {
 
 test_that("can read env vars when none set", {
   app <- local_temp_app()
-  addTestDeployment(app, "test1", envVars = NA_character_)
+  path <- addTestDeployment(app, "test1", envVars = NULL)
+  deps <- deployments(app, excludeOrphaned = FALSE)
+  expect_equal(deps$envVars, list(character()))
+  expect_false("envVars" %in% rownames(read.dcf(path)))
```
Comment (hadley) [inline tests/testthat/test-deployments.R:94]: @aronatkins this is where we test what happens when we deliberately create a deployment without `envVars` — we test that `envVars` is not written to the DCF, but it's still successfully read. I'll add something similar for `version`.

**Review by mslynch — APPROVED** [review]

File: tests/testthat/test-deployments.R (hunk 3 above)
Comment (aronatkins) [reply tests/testthat/test-deployments.R:94]: Thanks. I missed that this test was confirming the processing of an "old" DCF. Would you mind adding a comment / renaming the test to remind us that this is testing DCF compatibility upon upgrade?

**Pull request merged**